### DISCUSSION TITLE
design pattern

### DISCUSSION THREAD

**Discussion by akshaypat444** [keras-team/keras#19960] [General]
What design pattern Keras uses for library development?

**Comment by abhaskumarsinha** [comment]
Design Pattern??

I believe it is object oriented.

**Reply by akshaypat444** [reply]
Not the programming paradigm. I mean to say design pattern of object oriented program like structural patterns, behavioral pattern, creation patterns etc?

**Comment by aamijar** [comment]
Keras uses the builder design pattern to define models. The observer design pattern is also used for callbacks in Keras.
